feat: Add support for namespacedcloudprofiles

klocke-io opened the pull request:
**What this PR does / why we need it**:
Add support for namespacedcloudprofiles

**Which issue(s) this PR fixes**:
Fixes #1971

**Special notes for your reviewer**:
NSCP in create cluster dialog

<img width="927" height="1055" alt="image" src="https://github.com/user-attachments/assets/bee445ef-309e-448f-a4f0-6f454a74ec31" />

No more warnings because the cloudprofile is not understood
<img width="709" height="226" alt="image" src="https://github.com/user-attachments/assets/f60ffad3-1a67-4d1c-9d40-8751bbf7dfe2" />
<img width="667" height="307" alt="image" src="https://github.com/user-attachments/assets/1aa9919d-71f5-4202-9eae-148b82da482a" />
<img width="977" height="221" alt="image" src="https://github.com/user-attachments/assets/7189a6fd-be56-42b3-8641-a98e6b3cf817" />
<img width="981" height="202" alt="image" src="https://github.com/user-attachments/assets/034bc51d-81a4-4832-80c8-0b21f813ee0e" />

**Release note**:
<!--  Write your release note:
1. Enter your release note in the below block.
2. If no release note is required, just write "NONE" within the block.

Format of block header: <category> <target_group>
Possible values:
- category:       breaking|feature|bugfix|doc|other
- target_group:   user|operator|developer|dependency
-->
```feature user
Add namespacedcloudprofiles support
```

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Namespaced CloudProfile support across backend and frontend, including API endpoints and RBAC permissions.
  * Optional diff view for namespaced profiles to compare against parent profiles and rehydrate specs when needed.
  * Cloud profile selector and store UI updated to show regular and namespaced profiles uniformly; namespace-aware loading.

* **Tests**
  * Expanded test coverage for namespaced profiles, diffing, rehydration and authorization scenarios.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

klocke-io committed:
backend: add namespaced cloudprofile support with packages and chart updates

klocke-io committed:
frontend: support namespaced cloudprofiles including all-project scope

petersutter commented on charts/gardener-dashboard/charts/application/templates/dashboard/clusterrole.yaml:
```diff
@@ -50,6 +50,13 @@ rules:
+  - namespacedcloudprofiles
+  verbs:
+  - list
+  - watch
```
reminder (if not already done) we also need to adapt the gardener operator accordingly

klocke-io committed:
chore: fix fixture naming

klocke-io committed:
feat: add integrity verification for namespaced cloud profile rehydration

 Backend returns parentResourceVersion and cloudProfileSpecHash alongside diffs. Frontend verifies rehydrated specs against the hash, re-fetches drifted parents, and drops profiles that cannot be reconciled — surfacing errors to the user via the app store.

adenitiu commented:
Hi colleagues, any timeline by when can this PR be merged ?

petersutter commented:
Hi @adenitiu, this PR is on our top-priority list for PRs to be reviewed.

This PR fetches the namespaced cloud profiles when navigating to the shoot list for dedicated projects. This way, you can see potential warnings and information within the shoot list. The fetched NSCPs are, of course, also used for the create cluster dialog.

However, there is a limitation on the shoot list with the `ALL PROJECTS` filter, where no NSCPs are fetched. Hence, the information that is displayed in the list differs from the information displayed when you navigate to the respective project. We can also most likely not simply fetch all as this might blow up the memory on the frontend, so this is an open point and still under discussion how to handle this situation.

We also need to test it with data that is closer to a real landscape. Depending on the test results, we will then decide if we can go with the current implementation or if changes are needed.

petersutter commented:
/remove-lifecycle rotten
